prometheus-dns-registration-service/main.go

```diff
@@ -0,0 +1,358 @@
+}
+
+// checkAuth checks the AWS access
+func checkAuth(stsAPI stsiface.STSAPI) (string, error) {
+	input := &sts.GetCallerIdentityInput{}
+	output, err := stsAPI.GetCallerIdentity(input)
+	if err != nil {
+		return "", errors.Wrap(err, "checking AWS STS access – cannot get role ARN for current session")
+	}
+	iamRoleARN := *output.Arn
+	log.Debugf("role ARN for the current session is %s", iamRoleARN)
+	return iamRoleARN, nil
+}
+
+// getUsername get the username out of the AWS Lambda role arn
+func getUsername(iamRoleARN string) string {
+	usernameParts := strings.Split(iamRoleARN, "/")
+	if len(usernameParts) > 1 {
+		return usernameParts[len(usernameParts)-1]
+	}
+	return "iam-root-account"
+}
+
+// withEmbeddedToken create a new AWS token 
+func (c *clientConfig) withEmbeddedToken() (*clientConfig, error) {
+	clientConfigCopy := *c
+
+	log.Info("Generating token")
+
+	gen, err := token.NewGenerator()
+	if err != nil {
+		return nil, err
+	}
+
+	tok, err := gen.GetWithSTS(c.ClusterName, c.sts.(*sts.STS))
+	if err != nil {
+		return nil, err
+	}
+
+	x := c.Client.AuthInfos[c.ContextName]
+	x.Token = tok
+
+	log.Infof("Successfully generated token")
+	return &clientConfigCopy, nil
+}
+
+// newClientSetWithEmbeddedToken creates the new k8s client
+func (c *clientConfig) newClientSetWithEmbeddedToken() (*clientset.Clientset, *rest.Config, error) {
+	clientConfig, err := c.withEmbeddedToken()
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Info("Generating k8s client")
+	clientSet, config, err := clientConfig.newClientSet()
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Infof("Successfully generated client")
+	return clientSet, config, err
+}
+
+// newClientSet is used by newClientSetWithEmbeddedToken to create the k8s client
+func (c *clientConfig) newClientSet() (*clientset.Clientset, *rest.Config, error) {
+	clientConfig, err := clientcmd.NewDefaultClientConfig(*c.Client, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, clientConfig, err
+	}
+
+	client, err := clientset.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, clientConfig, err
+	}
+	return client, clientConfig, err
+}
+
+// getRoute53Records is used to get the existing Route53 Records
+func getRoute53Records(hostedZoneID string) ([]string, error) {
+	sess, err := session.NewSession(&aws.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Route53 service client
+	svc := route53.New(sess)
+
+	recordSets, err := svc.ListResourceRecordSets(&route53.ListResourceRecordSetsInput{
+		HostedZoneId: aws.String(hostedZoneID),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	matchingRecords := getMatchingRecords(recordSets.ResourceRecordSets)
+
+	return matchingRecords, nil
+}
+
+// getMatchingRecords is used to get only Prometheus related Route53 records.
+func getMatchingRecords(recordSets []*route53.ResourceRecordSet) []string {
+	matchingRecords := []string{}
+	fmt.Println(recordSets)
```

Comment:
Why not use log here?

Reply:
Missed that thanks :) It was used for debugging, not needed there.